docs/developer/sdk_from_swagger_and_typespec.md

```diff
@@ -0,0 +1,195 @@
+class Person(_serialization.Model):
+    _attribute_map = {
+        "name": {"key": "name", "type": "str"},
+        "properties_name": {"key": "properties.name", "type": "str"},
+    }
+
+    def __init__(self, *, name: Optional[str] = None) -> None:
+        ...
+
+msrest_model = Person(name="xxx", properties_name="properties_name")
+print(msrest_model.name)
+print(msrest_model.properties_name) # A
+print(msrest_model.properties.name) # equivalent to A
+```
+
+Due to inconsistent usage of flattening in some Swagger specifications, property names can become unwieldy and user-unfriendly. For this reason, **DPG models do not support flattening**.
+
+#### Flattening Compatibility Note
+
+For legacy SDKs generated from Swagger that are migrated to TypeSpec, we've designed a [compatibility mechanism](https://azure.github.io/typespec-azure/docs/howtos/generate-client-libraries/07types/#flattening) to minimize breaking changes. However, for deeply nested flattened properties, code updates may be required:
+
+```python
+# Msrest model
+msrest_model = Model(...)
+
+print(msrest_model.properties_name) # A
+print(msrest_model.properties.name) # equivalent to A
+print(msrest_model.properties_properties_name) # B
+print(msrest_model.properties.properties.name) # equivalent to B
+
+# After migration to TypeSpec
+dpg_model = Model(...)
+print(dpg_model.properties_name) # A, backwards compatible but not recommended
+print(dpg_model.properties.name) # equivalent to A
+print.dpg_model.properties_properties_name) # no longer works
+print(dpg_model.properties.properties.name) # recommended approach after migration
+```
+
+### Additional Properties
+
+#### Additional Properties in Msrest Models
+To support [additional properties](https://www.apimatic.io/openapi/additionalproperties), Msrest models include an `additional_properties` parameter:
+
+```python
+msrest_model = Model(additional_properties={"hello": "world"})
+print(msrest_model.as_dict()) # output is `{"hello": "world"}`
+```
+
+#### Additional Properties in DPG Models
+DPG models inherently support additional properties through dictionary-like behavior:
+
+```python
+dpg_model = Model({"hello": "world"})
+# or
+dpg_model = Model()
+dpg_model.update({"hello": "world"})
+# or 
+dpg_model = Model()
+dpg_model["hello"] = "world"
+
+print(dpg_model) # output is `{"hello": "world"}`
+```
+
+## Query/Header Parameters in Operations
+
+Query and header parameters in Swagger-generated SDKs are positional, while in TypeSpec-generated SDKs they are keyword-only:
+
+```python
+# Swagger SDK
+client.operation("header", "query") # A
+client.operation(header_parameter="header", query_parameter="query") # equivalent to A
+
+# After migration to TypeSpec
+client.operation("header", "query") # no longer works
+client.operation(header_parameter="header", query_parameter="query") # correct approach
+```
+
+## File Name Changes
+After migration, some internal file names change, but these changes do not affect SDK users:
```

Comment:
i would make an ote that these files were internal to begin with. otherwise this all looks great, thanks @msyyc!